File: b32/64to32.py
#!/usr/bin/env python
#
# Written by duck
# http://forum.i2p/viewtopic.php?t=4367
#
import base64, hashlib, sys

if len(sys.argv) != 2:
    print 'Usage: 64to32.py <base64key>'
    sys.exit(1)

key = sys.argv[1]
raw_key = base64.b64decode(key, '-~')
hash = hashlib.sha256(raw_key)
base32_hash = base64.b32encode(hash.digest())
print base32_hash.lower().replace('=', '')+'.b32.i2p'

File: b32/64to32.pl
#!/usr/bin/perl
#
# zzz 2020-05 CC0
#


use strict;
use MIME::Base64;
use Convert::Base32;
use Digest::SHA qw(sha256);

my $num_args = $#ARGV + 1;
if ($num_args != 1) {
    print "Usage: 64to32.pl base64\n";
    exit;
}

my $bb64 = $ARGV[0];
my $b64 = $bb64;
$b64 =~ s/-/+/g;
$b64 =~ s/~/\//g;
my $decoded = decode_base64($b64);
my $hash=sha256($decoded);
my $encoded = encode_base32($hash);
#print "Base64: " . $bb64 . "\n";
print "Base32: " . $encoded . ".b32.i2p\n";
